imagebuildah/build.go

```diff
@@ -647,58 +667,52 @@ func (b *Executor) Commit(ib *imagebuilder.Builder) (err error) {
 }
 
 // Build takes care of the details of running Prepare/Execute/Commit/Delete
-// over each of the one or more parsed Dockerfiles.
-func (b *Executor) Build(ib *imagebuilder.Builder, node []*parser.Node) (err error) {
-	if len(node) == 0 {
-		return errors.Wrapf(err, "error building: no build instructions")
-	}
-	first := node[0]
-	from, err := ib.From(first)
-	if err != nil {
-		logrus.Debugf("Build(first.Children=%#v)", first.Children)
-		return errors.Wrapf(err, "error determining starting point for build")
-	}
-	if err = b.Prepare(ib, first, from); err != nil {
-		return err
-	}
-	defer b.Delete()
-	for _, this := range node {
-		if err = b.Execute(ib, this); err != nil {
+// over each of the one or more parsed Dockerfiles and stages.
+func (b *Executor) Build(stages imagebuilder.Stages) error {
+	if len(stages) == 0 {
+		errors.New("error building: no stages to build")
+	}
+	var stageExecutor *Executor
+	for _, stage := range stages {
+		stageExecutor = b.withName(stage.Name)
```

Comment:
This should take the index of the stage as an argument, since whether a name for the stage is specified with `AS` or not, its contents still need to be accessible by referring to the stage using its numeric index, and we want to know where a given `Executor` is in the list of them.  Converting the number to a string in `withName()` and using it as another key for adding `copied` to the `named` map, since the values are pointers, should be fine.

Reply:
Done. Thanks for catching this.

I checked the behaviour of `docker` and it does accept both indexes and names. On the other hand, `imagebuilder` does not. We might report this.